pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -243,42 +249,76 @@ func (c *DeploymentConfigController) reconcileDeployments(existingDeployments *k
 		if isActiveDeployment {
 			newReplicaCount = activeReplicas
 		}
-		if config.Spec.Test {
-			glog.V(4).Infof("Deployment config %q is test and deployment %q will be scaled down", deployutil.LabelForDeploymentConfig(config), deployutil.LabelForDeployment(&deployment))
+		if copied.Spec.Test {
+			glog.V(4).Infof("Deployment config %q is test and deployment %q will be scaled down", deployutil.LabelForDeploymentConfig(copied), deployutil.LabelForDeployment(&deployment))
 			newReplicaCount = 0
 		}
 		lastReplicas, hasLastReplicas := deployutil.DeploymentReplicas(&deployment)
 		// Only update if necessary.
 		if !hasLastReplicas || newReplicaCount != oldReplicaCount || lastReplicas != newReplicaCount {
-			deployment.Spec.Replicas = newReplicaCount
-			deployment.Annotations[deployapi.DeploymentReplicasAnnotation] = strconv.Itoa(int(newReplicaCount))
-			_, err := c.kubeClient.ReplicationControllers(deployment.Namespace).Update(&deployment)
+			copied, err := rcDeepCopy(deployment)
```

Comment:
`deploymentCopy`?

I've also used `sharedDeployment` to mean, "object that isn't mine".  I don't care which, but you can't shadow your own var.